src/overlay/Peer.cpp

```diff
@@ -743,10 +750,22 @@ Peer::recvTransaction(StellarMessage const& msg)
             // record that this peer sent us this transaction
             mApp.getOverlayManager().recvFloodedMsg(msg, shared_from_this());
 
+            size_t size = xdr::xdr_argpack_size(msg);
             if (recvRes == TransactionQueue::AddResult::ADD_STATUS_PENDING)
             {
                 // if it's a new transaction, broadcast it
                 mApp.getOverlayManager().broadcastMessage(msg);
+
+                mUniqueBytesRecv.Mark(size);
```

Comment:
this doesn't seem to be the right place: if we receive a stream of bad (expired) transactions, we're not going to count them against "bad" data.

Easiest might be to just lookup the message in the floodmap, and if it was already there, count it as "waste" (duplicate/not needed) *before* feeding that transaction to herder.